File: bootstrap/src/main/java/net/dontdrinkandroot/wicket/bootstrap/component/button/RepeatingButtonToolbar.java
/*
 * Copyright (C) 2012-2017 Philip Washington Sorst <philip@sorst.net>
 * and individual contributors as indicated
 * by the @authors tag.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package net.dontdrinkandroot.wicket.bootstrap.component.button;

import net.dontdrinkandroot.wicket.behavior.CssClassAppender;
import net.dontdrinkandroot.wicket.bootstrap.css.BootstrapCssClass;
import org.apache.wicket.markup.html.panel.GenericPanel;
import org.apache.wicket.markup.repeater.RepeatingView;
import org.apache.wicket.model.IModel;

/**
 * @param <T> Type of the model object.
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public abstract class RepeatingButtonToolbar<T> extends GenericPanel<T>
{
    public RepeatingButtonToolbar(String id)
    {
        super(id);
    }

    public RepeatingButtonToolbar(String id, IModel<T> model)
    {
        super(id, model);
    }

    @Override
    protected void onInitialize()
    {
        super.onInitialize();

        this.add(new CssClassAppender(BootstrapCssClass.BTN_TOOLBAR));

        RepeatingView buttonGroupView = new RepeatingView("buttonGroup");
        this.populateButtonGroups(buttonGroupView);
        this.add(buttonGroupView);
    }

    protected abstract void populateButtonGroups(RepeatingView buttonGroupView);
}

File: bootstrap/src/main/kotlin/net/dontdrinkandroot/wicket/bootstrap/component/button/RepeatingButtonToolbar.kt
package net.dontdrinkandroot.wicket.bootstrap.component.button

import net.dontdrinkandroot.wicket.behavior.CssClassAppender
import net.dontdrinkandroot.wicket.bootstrap.css.BootstrapCssClass
import org.apache.wicket.markup.html.panel.GenericPanel
import org.apache.wicket.markup.repeater.RepeatingView
import org.apache.wicket.model.IModel

/**
 * @param <T> Type of the model object.
 */
abstract class RepeatingButtonToolbar<T>(id: String, model: IModel<T>? = null) : GenericPanel<T>(id, model) {

    override fun onInitialize() {
        super.onInitialize()
        this.add(CssClassAppender(BootstrapCssClass.BTN_TOOLBAR))
        val buttonGroupView = RepeatingView("buttonGroup")
        populateButtonGroups(buttonGroupView)
        this.add(buttonGroupView)
    }

    protected abstract fun populateButtonGroups(buttonGroupView: RepeatingView)
}